File: Python/7py 2/tidyNumber.py
"""
|--------------------------------------------------------------------------
| Tidy Number (Special Numbers Series #9)
|--------------------------------------------------------------------------
|
"""

def tidyNumber(n):
   return list(str(n)) == sorted(list(str(n)))


def tidyNumber(n):
   return n == int(''.join(sorted(str(n))))


def tidyNumber(n):
   n = str(n)
   for i in range(0, len(n) - 1):
      if n[i] > n[i + 1]:
         return False
   return True


def tidyNumber(n):
   a = str(n)
   index = 0
   while index<len(a)-1:
      if int(a[index])>int(a[index+1]):
         return False
      index +=1
   return True


print(tidyNumber(12), True)
print(tidyNumber(102), False)
print(tidyNumber(9672), False)
print(tidyNumber(2789), True)

File: JavaScript/7js 3/tidyNumber.js
/*
|--------------------------------------------------------------------------
| Tidy Number (Special Numbers Series #9)
|--------------------------------------------------------------------------
|
*/

function tidyNumber(n) {
   return String(n).split('').join('') == String(n).split('').sort((a, b) => a - b).join('');
}


function tidyNumber(n) {
   return [...n += ""].sort().join `` == n
}


function tidyNumber(n) {
   let s = n.toString();
   for (let i = 0; i < s.length - 1; i++) {
      if (s[i] > s[i + 1]) return false;
   }
   return true;
}


function tidyNumber(n) {
   let m = 9;
   while (n) {
      if (n % 10 > m)
         return false;
      m = n % 10;
      n = n / 10 | 0;
   }
   return true;
}


console.log(tidyNumber(12), true);
console.log(tidyNumber(102), false);
console.log(tidyNumber(9672), false);
console.log(tidyNumber(2789), true);
console.log(tidyNumber(2335), true);